Mage.Sets/src/mage/cards/s/StormOfSouls.java

```diff
@@ -0,0 +1,153 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.ExileSpellEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class StormOfSouls extends CardImpl {
+    public StormOfSouls(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{4}{W}{W}");
+
+        // Return all creature cards from your graveyard to the battlefield.
+        // Each of them is a 1/1 Spirit with flying in addition to its other types.
+        this.getSpellAbility().addEffect(new StormOfSoulsReturnEffect());
+
+        // Exile Storm of Souls.
+        this.getSpellAbility().addEffect(new ExileSpellEffect().concatBy("<br>"));
+    }
+
+    private StormOfSouls(final StormOfSouls card) { super(card); }
+
+    @Override
+    public StormOfSouls copy() { return new StormOfSouls(this); }
+}
+
+class StormOfSoulsReturnEffect extends OneShotEffect {
+    public StormOfSoulsReturnEffect() {
+        super(Outcome.Benefit);
+        staticText = "Return all creature cards from your graveyard to the battlefield." +
+                "Each of them is a 1/1 Spirit with flying in addition to its other types.";
+    }
+
+    private StormOfSoulsReturnEffect(final StormOfSoulsReturnEffect effect) { super(effect); }
+
+    @Override
+    public StormOfSoulsReturnEffect copy() { return new StormOfSoulsReturnEffect(this); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+
+        if (player == null) { return false; }
+
+        Set<Card> creatureCardsToBeMovedFromGraveyard = player.getGraveyard().getCards(StaticFilters.FILTER_CARD_CREATURE, game);
+        if (creatureCardsToBeMovedFromGraveyard == null) { return false; }
+
+        boolean allCardsMoved;
+        if (creatureCardsToBeMovedFromGraveyard.isEmpty()) {
+            allCardsMoved = true;
+        } else {
+            // Returns true if all cards were moved.
+            allCardsMoved = player.moveCards(creatureCardsToBeMovedFromGraveyard, Zone.BATTLEFIELD, source, game);
```

Comment:
That function returns true if *any* cards were moved.  It doesn't mean they were all moved.  So your below logic to figure out which cards to be turned into spirits needs to always be done.

Reply:
Changing.

What should this `apply()` call return? I'm still not clear on it.

Reply:
I usually have it return true if it completes any part of the effect.  So here, return false if no creatures get moved to the battlefield.  Otherwise return true.